### PR TITLE
feat: add nonce indicator on pending transactions when viewing an address (#622)

### PR DESCRIPTION
**Pull request by linden**
Adds a indicator next to pending transactions, reasons detailed here (#622).

### CODE REVIEW

**Comment by linden** [comment]
I've got 3 options for how this could be displayed:

A:
image
Simple, although looks awkward when compared to non-pending transactions.

B: 
image
Easy to associate with a pending transaction.

C:
image
Blends in, would need to add some sort of hover-over description.

Any feedback on what you all think fits the best?

**Comment by kyranjamie** [comment]
Great stuff, thanks for the contribution. Def C with a tooltip. Maybe the number styleized as e.g. `17n`? 🤔

**Comment by 314159265359879** [comment]
Thanks for picking this up @linden here are my thoughts:

A is most descriptive, but I like C most because it least noticeable and I think that is fine in most cases. I think it is clever to put in a line we were already using as to not make the increase the height required per item shown. 

What about C including the word _nonce_? 
_Pending - 0xa31d...f886 - nonce: 14_

**Comment by 314159265359879** [comment]
What Kyran says even better `14n`, the hover will do the rest.

**Comment by fbwoolf** [comment]
> Great stuff, thanks for the contribution. Def C with a tooltip. Maybe the number styleized as e.g. `17n`? 🤔

I like this idea bc on the home page there is limited space across. I don't think spelling out `nonce` is necessary? I would say `C` is best and add the `n`

**Comment by linden** [comment]
Thanks for the feedback! Here's what it looks like with the tooltip and the `n`.

https://user-images.githubusercontent.com/70739041/150647564-3f206042-2354-4dfe-a658-b3786e21c003.mov

**Review by 314159265359879 — APPROVED** [review]
LGTM

**Comment by Hero-Gamer** [comment]
> > Great stuff, thanks for the contribution. Def C with a tooltip. Maybe the number styleized as e.g. `17n`? 🤔
> 
> I like this idea bc on the home page there is limited space across. I don't think spelling out `nonce` is necessary? I would say `C` is best and add the `n`

I'll speak from users perspective if I may?

as much as I like "n"
I think spelled out "Nonce" is easier for the less advanced users, which there are a lot across the ecosystem. One less thing for us having to explain. There should be plenty of space on the page (majority uses desktop to access Explorer)

I'd also like to suggest add "Fees" to the Explorer. Both Explorer and Hiro wallet if possible. 
Sorry!

I've made the comment on separate issues: 
https://github.com/hirosystems/explorer/issues/622
https://github.com/hirosystems/stacks-wallet-web/issues/2129

**Review by landitus — APPROVED** [review]
Even though I see good points for making the nonce more discoverable for new users, I'm thinking it's best to keep the line items as simple as possible given they repeat multiple times.  Right now, each information doesn't have a label, which is a good decision to avoid repeating text and crowding the interface. 

Having the tooltip as shown [here](https://github.com/hirosystems/explorer/pull/624#issuecomment-1019306403) should do the trick now.

For adding more information, such as fees, it will be more challenging (since there are no labels to explain what each piece of information means). I think then it may be time to review the line item's design a bit further.

File: src/components/transaction-item.tsx [hunk 1]
```diff
@@ -205,9 +219,14 @@ const LargeVersion = React.memo(
...
               </Caption>
               {'·'}
-              <Caption mt="1px" ml="6px">
+              <Caption mt="1px" ml="6px" mr={isPending && "6px"}>
                 {truncateMiddle(tx.tx_id, 4)}
               </Caption>
+
+              { isPending &&
+                <Nonce ml="6px" nonce={tx.nonce}>
+                </Nonce>
```
Comment (He1DAr) [inline src/components/transaction-item.tsx:228]: nit:
```suggestion
                <Nonce ml="6px" nonce={tx.nonce}/>
```

File: src/components/transaction-item.tsx [hunk 2]
```diff
@@ -158,6 +159,19 @@ const Timestamp: React.FC<BoxProps & { tx: Transaction | MempoolTransaction }> =
   }
 );
 
+const Nonce: React.FC<BoxProps & { nonce: Number, isHovered?: boolean }> = React.memo(
```
Comment (He1DAr) [inline src/components/transaction-item.tsx:162]: Does this component only need `nonce`? 
```suggestion
const Nonce: React.FC<{ nonce: Number }> = React.memo(
```

**Review by He1DAr — COMMENTED** [review]
LGTM! left a couple of small comments. Could you resolve CI issues?

File: src/components/transaction-item.tsx (hunk 2 above)
Comment (linden) [reply src/components/transaction-item.tsx:162]: Probably, I'm new to React so I was really winging this PR.

**Comment by linden** [comment]
Looks ready to me, anything else needed @He1DAr?

**Comment by 314159265359879** [comment]
This is brilliant, I love it! 
Makes it so much easier to troubleshoot nonce issues. 
image

**Pull request closed without merging**